src/main/scala/dpla/ingestion3/enrichments/date/DateBuilder.scala

```diff
@@ -1,109 +1,53 @@
+    * @param originalSourceDate Un-enriched date value
+    * @return EdmTimeSpan
     */
-  def buildEdmTimeSpan(date: String): EdmTimeSpan = {
-    val normalizedDate = normalizeDate(date)
-    DateBuilderPatterns.edtfPatterns.foreach(edtf => {
-      // Perform a simple length comparision
-      if(edtf.parser.length == normalizedDate.length) {
-        // Get the parser
-        val parser = DateFormatCache.get(edtf.parser)
-
-        Try { parser.parse(normalizedDate) } match {
-          case Success(parsedDate) =>
-            val formatter = DateFormatCache.get(edtf.format)
-            val formattedDate = formatter.format(parsedDate)
-
-            return EdmTimeSpan(
-              originalSourceDate = Option(date),
-              prefLabel = Option(formattedDate),
-              begin = Option(formattedDate), // FIXME this will not pass muster for date ranges
-              end = Option(formattedDate) // FIXME this will not pass muster for date ranges
-            )
-          case Failure(_) => None
-        }
-      }
-    })
-    stringOnlyTimeSpan(date)
+  def generateBeginEnd(originalSourceDate: ZeroToOne[String]): EdmTimeSpan = {
+    originalSourceDate match {
+      case None => emptyEdmTimeSpan
+      case Some(d) =>
+        EdmTimeSpan(
+          originalSourceDate = Some(d),
+          prefLabel = Some(d),
+          begin = createBegin(d),
+          end = createEnd(d)
+        )
+    }
   }
 
   /**
-    * Normalizes a provided date string to facilitate pattern matching
-    * - Replace all supported delimiters (forward slash, dash) with whitespace
+    * Identifies a begin date
     *
-    * @param date Date string
-    * @return Normalized date string
+    * @param str Original date value
+    * @return Option[String]
     */
-  def normalizeDate(date: String): String = date.replaceAll("[\\/-]", " ")
-}
-
-
-object DateFormatCache {
-
-  private val cache = new ThreadLocal[collection.mutable.Map[String, SimpleDateFormat]]
-  cache.set(collection.mutable.Map[String, SimpleDateFormat]())
+  def createBegin(str: String): Option[String] = str match {
+    case `str` if str.matches("[0-9]{4}") => Some(str)
+    case `str` if str.matches("[0-9]{4}-[0-9]{4}") => str.split("-").headOption
+    case _ => None
+  }
 
-  def get(formatString: String): SimpleDateFormat = {
-    val localCache = cache.get()
-    localCache.get(formatString) match {
-      case None =>
-        val simpleDateFormat = new SimpleDateFormat(formatString)
-        simpleDateFormat.setTimeZone(TimeZone.getTimeZone("EST"))
-        simpleDateFormat.setLenient(true)
-        localCache.put(formatString, simpleDateFormat)
-        simpleDateFormat
-      case Some(simpleDateFormat) => simpleDateFormat
+  /**
+    * Identifies an end date
+    *
+    * @param str Original date value
+    * @return Option[String]
+    */
+  def createEnd(str: String): Option[String] = str match {
+      case `str` if str.matches("[0-9]{4}") => Some(str)
```

Comment:
Won't this match both cases? You should probably either stick anchors in this regular expression for the start/end, or swap the order of these two.

Reply:
It's weird that this would pass the test, though. Am I wrong?